### DISCUSSION TITLE
What is the 'key' attribute of Vscode.TextEditorDecorationType.create?

### DISCUSSION THREAD

**Discussion by aqjune** [ocamllabs/vscode-ocaml-platform#1372] [Q&A] [answered]
Hi all,

Can someone explain what the 'key' attribute of Vscode.TextEditorDecorationType.create is? I would appreciate a lot if an example for using 'key' is provided as well.

**Comment by mlantas** [comment, marked as answer]
You can reference the official VS Code API documentation: https://code.visualstudio.com/api/references/vscode-api

The [documentation for TextEditorDecorationType](https://code.visualstudio.com/api/references/vscode-api#TextEditorDecorationType) says that `key` is the "Internal representation of the handle."
More importantly, it says the following:
> To get an instance of a `TextEditorDecorationType` use [createTextEditorDecorationType](https://code.visualstudio.com/api/references/vscode-api#window.createTextEditorDecorationType).

That means you don't have to set the `key` value yourself. However, since we don't have bindings for the `createTextEditorDecorationType` function yet, that means you will have to add the binding yourself (inside the `Window` module). That function has a parameter of type [DecorationRenderOptions](https://code.visualstudio.com/api/references/vscode-api#DecorationRenderOptions) that also needs bindings.

If it's helpful, I found a TypeScript example of using decorators here: https://github.com/microsoft/vscode-extension-samples/blob/main/decorator-sample/USAGE.md

**Comment by aqjune** [comment]
Thanks, it helped a lot. :)
